zarr/hierarchy.py

```diff
@@ -542,6 +545,59 @@ def visititems(self, func):
         base_len = len(self.name)
         return self.visitvalues(lambda o: func(o.name[base_len:].lstrip("/"), o))
 
+    def tree(self):
+        """Provide a ``print`-able display of the hierarchy.
+
+        Examples
+        --------
+        >>> import zarr
+        >>> g1 = zarr.group()
+        >>> g2 = g1.create_group('foo')
+        >>> g3 = g1.create_group('bar')
+        >>> g4 = g3.create_group('baz')
+        >>> g5 = g3.create_group('quux')
+        >>> d1 = g5.create_dataset('baz', shape=100, chunks=10)
+        >>> print(g1.tree())
+        /
+         ├── bar
+         │   ├── baz
+         │   └── quux
+         │       └── baz[...]
+         └── foo
+        >>> print(g3.tree())
+        bar
+         ├── baz
+         └── quux
+             └── baz[...]
```

Comment:
Opted to try displaying the datasets like this. Please let me know what you think.